### ISSUE TITLE
RTDETR: map class names of pretrained model to same names as in YOLOv8

### ISSUE THREAD

**Issue by ruzko** [ultralytics/ultralytics#9238]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Both model types are presumably trained on COCO, with COCO classes and COCO class names.
Verified with a few different images that the detected class id is identical with both model types.
I understand from #4092 that there's some uncertainty related to models trained on various datasets, but since the pretrained models downloaded and run through Ultralytics are trained on the same dataset, don't they have the same circumstances?

```python
>>> from ultralytics import RTDETR
>>> model_rtdetr = RTDETR('models/rtdetr-l.pt')
>>> model_rtdetr.names
{0: '0', 1: '1', 2: '2', 3: '3', 4: '4', 5: '5', 6: '6', 7: '7', 8: '8', 9: '9', 10: '10', 11: '11', 12: '12', 13: '13', 14: '14', 15: '15', 16: '16', 17: '17', 18: '18', 19: '19', 20: '20', 21: '21', 22: '22', 23: '23', 24: '24', 25: '25', 26: '26', 27: '27', 28: '28', 29: '29', 30: '30', 31: '31', 32: '32', 33: '33', 34: '34', 35: '35', 36: '36', 37: '37', 38: '38', 39: '39', 40: '40', 41: '41', 42: '42', 43: '43', 44: '44', 45: '45', 46: '46', 47: '47', 48: '48', 49: '49', 50: '50', 51: '51', 52: '52', 53: '53', 54: '54', 55: '55', 56: '56', 57: '57', 58: '58', 59: '59', 60: '60', 61: '61', 62: '62', 63: '63', 64: '64', 65: '65', 66: '66', 67: '67', 68: '68', 69: '69', 70: '70', 71: '71', 72: '72', 73: '73', 74: '74', 75: '75', 76: '76', 77: '77', 78: '78', 79: '79'}


>>> from ultralytics import YOLO
>>> model_yolo = YOLO('models/yolov8x.pt')
>>> model_yolo.names
{0: 'person', 1: 'bicycle', 2: 'car', 3: 'motorcycle', 4: 'airplane', 5: 'bus', 6: 'train', 7: 'truck', 8: 'boat', 9: 'traffic light', 10: 'fire hydrant', 11: 'stop sign', 12: 'parking meter', 13: 'bench', 14: 'bird', 15: 'cat', 16: 'dog', 17: 'horse', 18: 'sheep', 19: 'cow', 20: 'elephant', 21: 'bear', 22: 'zebra', 23: 'giraffe', 24: 'backpack', 25: 'umbrella', 26: 'handbag', 27: 'tie', 28: 'suitcase', 29: 'frisbee', 30: 'skis', 31: 'snowboard', 32: 'sports ball', 33: 'kite', 34: 'baseball bat', 35: 'baseball glove', 36: 'skateboard', 37: 'surfboard', 38: 'tennis racket', 39: 'bottle', 40: 'wine glass', 41: 'cup', 42: 'fork', 43: 'knife', 44: 'spoon', 45: 'bowl', 46: 'banana', 47: 'apple', 48: 'sandwich', 49: 'orange', 50: 'broccoli', 51: 'carrot', 52: 'hot dog', 53: 'pizza', 54: 'donut', 55: 'cake', 56: 'chair', 57: 'couch', 58: 'potted plant', 59: 'bed', 60: 'dining table', 61: 'toilet', 62: 'tv', 63: 'laptop', 64: 'mouse', 65: 'remote', 66: 'keyboard', 67: 'cell phone', 68: 'microwave', 69: 'oven', 70: 'toaster', 71: 'sink', 72: 'refrigerator', 73: 'book', 74: 'clock', 75: 'vase', 76: 'scissors', 77: 'teddy bear', 78: 'hair drier', 79: 'toothbrush'}
```

### Use case

It would be nice to be able to swap between pretrained models downloaded via ultralytics, and still have human readable class names :)
I'm making a FLOSS video surveillance solution using Ultralytics. While the project will eventually use models finetuned on a specific subset of classes (with human readable class names), for now it uses pretrained RTDETR models with default number-only class names.
For demos especially, it's good to be able to show human readable labels for many different classes.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by Burhan-Q** [comment]
@ruzko this was updated recently. Try deleting your local copy and then run `model_rtdetr = RTDETR('rtdetr-l.pt')` and it should download a new copy that have the COCO names

**Comment by ruzko** [comment]
I can confirm that this works with a newly downloaded model. Great, thanks!
